release.yaml

```diff
@@ -21,267 +21,320 @@ internal:
+          cmd: |
+            comby -matcher ".generic" -in-place "version: \":[~\d+\.\d+\.\d+]\"" 'version: "{{inputs.server.tag}}"' -f charts/sourcegraph/Chart.yaml
+            comby -matcher ".generic" -in-place "version: \":[~\d+\.\d+\.\d+]\"" 'version: "{{inputs.server.tag}}"' -f charts/sourcegraph/examples/subchart/Chart.yaml
+            comby -matcher ".generic" -in-place "version: \":[~\d+\.\d+\.\d+]\"" 'version: "{{inputs.server.tag}}"' -f charts/sourcegraph-executor/dind/Chart.yaml
+            comby -matcher ".generic" -in-place "version: \":[~\d+\.\d+\.\d+]\"" 'version: "{{inputs.server.tag}}"' -f charts/sourcegraph-executor/k8s/Chart.yaml
+            comby -matcher ".generic" -in-place "version: \":[~\d+\.\d+\.\d+]\"" 'version: "{{inputs.server.tag}}"' -f charts/sourcegraph-migrator/Chart.yaml
+         
+        - name: "chart:appVersion"
+          cmd: |
+            comby -matcher ".generic" -in-place "appVersion: \":[~\d+\.\d+\.\d+]\"" 'appVersion: "{{inputs.server.tag}}"' -f charts/sourcegraph/Chart.yaml
+            comby -matcher ".generic" -in-place "appVersion: \":[~\d+\.\d+\.\d+]\"" 'appVersion: "{{inputs.server.tag}}"' -f charts/sourcegraph-executor/dind/Chart.yaml
+            comby -matcher ".generic" -in-place "appVersion: \":[~\d+\.\d+\.\d+]\"" 'appVersion: "{{inputs.server.tag}}"' -f charts/sourcegraph-executor/k8s/Chart.yaml
+            comby -matcher ".generic" -in-place "appVersion: \":[~\d+\.\d+\.\d+]\"" 'appVersion: "{{inputs.server.tag}}"' -f charts/sourcegraph-migrator/Chart.yaml
+          
+        - name: "helm:docs"
+          cmd: ./scripts/helm-docs.sh
+
+        - name: "git:branch"
+          cmd: |
+            set -eu
+            branch="internal/release-{{version}}"
+            echo "Creating branch $branch"
+            git switch -c $branch
+
+        - name: "git:commit"
+          cmd: |
+            git commit -am 'release_major: {{version}}' -m '{{config}}'
+
+        - name: "git:push"
+          cmd: |
+            branch="internal/release-{{version}}"
+            git push origin ${branch}
+
+        - name: "github:pr"
+          cmd: |
+            set -eu
+            gh pr create \
+              --fill \
+              --draft \
+              --title "(internal) release_major: build {{version}}" \
+              --body "Test plan: automated release PR, CI will perform additional checks"
+            echo "🚢 Please check the associated CI build to ensure the process completed".
+  
   finalize:
     steps:
+      - name: 'Register on release registry'
+        cmd: |
+          echo "Registering internal sourcegraph/sourcegraph {{version}} release on release registry"
+          body=$(curl -s --fail-with-body -X POST "https://releaseregistry.sourcegraph.com/v1/releases" -H "Content-Type: application/json" -H "Authorization: ${RELEASE_REGISTRY_TOKEN}" -d '{
+              "name": "helm",
+              "version": "{{version}}",
+              # "git_sha": "${COMMIT_SHA}"
```

Comment:
@jhchabran I noticed the `COMMIT_SHA` variable isn't available in this step on Buildkite. How can I expose this?

![CleanShot 2024-04-24 at 16 45 24@2x](https://github.com/sourcegraph/deploy-sourcegraph-helm/assets/25608335/dc47793e-4d1a-4b35-8124-14656daf9e18)

Reply:
Since the repo is checked out on buildkite we could probably run `export GIT_SHA=$(git rev-parse HEAD)` in a previous step. There's also `BUILDKITE_COMMIT` but I'm not certain that's the hash we actually want.

Reply:
`BUILDKITE_COMMIT` can take `HEAD` as a value, so the `git rev-parse HEAD` is the best course of action here.